### PR TITLE
feat: Add `ReplaceObsoletesStep`

### PR DESCRIPTION
**Pull request by Pankraz76**
feat: Add `ReplaceObsoletesStep`

This PR introduces a new step to remove common boilerplate code, such as:
- Redundant declarations reimplementing Java defaults
- Unnecessary explicit language defaults
- Other similar redundancies

**Motivation:**  
Discovered during the review of:
- #2524 
  - This [comment](https://github.com/diffplug/spotless/pull/2524#issuecomment-3002059471) highlighting the need for such cleanup.

**Benefits:**
- Reduces noise in codebases
- Eliminates unnecessary maintenance of obvious defaults
- Follows DRY and CoC principles

### CODE REVIEW

**Comment by Pankraz76** [comment]
kindly request review @iddeepak.

**Comment by iddeepak** [comment]
Add [here](https://github.com/Pankraz76/spotless/blob/5b74aa0018053d15f5ff9f8504da675f8d97861e/CHANGES.md?plain=1#L12)
plugin-gradle [here](https://github.com/Pankraz76/spotless/blob/5b74aa0018053d15f5ff9f8504da675f8d97861e/plugin-gradle/CHANGES.md?plain=1#L5)
plugin-maven [here](https://github.com/Pankraz76/spotless/blob/5b74aa0018053d15f5ff9f8504da675f8d97861e/plugin-maven/CHANGES.md?plain=1#L5)


### Added
`ReplaceObsoletesStep` removes redundant default initializations ([#2530](https://github.com/diffplug/spotless/pull/2530))

File: plugin-maven/src/test/java/com/diffplug/spotless/maven/java/ReplaceObsoletesStepTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		writePomWithJavaSteps("<replaceObsoletes/>");
+
+		String path = "src/main/java/test.java";
+		setFile(path).toResource("java/replaceobsoletes/IntInitializersPre.test");
+		mavenRunner().withArguments("spotless:apply").runNoError();
+		assertFile(path).sameAsResource("java/replaceobsoletes/IntInitializersPost.test");
+	}
+
+	@Test
+	@Disabled("feature envy: (edge case/hard to implement) having a dedicated method")
```
Comment (iddeepak) [inline plugin-maven/src/test/java/com/diffplug/spotless/maven/java/ReplaceObsoletesStepTest.java:95]: Thanks for adding this. Since it's currently disabled, maybe a follow-up issue should be created ?

**Comment by Pankraz76** [comment]
thx for feedback.

**Review by iddeepak — APPROVED** [review]

**Comment by iddeepak** [comment]
> thx for feedback.

Thank you!

**Comment by nedtwigg** [comment]
Spotless is primarily a switchboard for existing formatters. Sometimes we merge in a step where the implementation is inside our repository, but that tends to cause problems with backward compatibility - Spotless has pretty strict backwards-compatibility. If the step implementation lives in a separate jar, that gives the step a lot more flexibility to change, and Spotless users can choose to pin an older version of the step's behavior but still get new Gradle/Maven support from Spotless if they want.

In my view, this step has some great cleanups (redundant `static` for example), and some that I disagree with (I prefer to always initialize fields, I think Kotlin designed this better than Java, I would prefer a rule that does the opposite of this one).

I'm very happy to merge support for this step whether I agree with it or not, but I think it should live in a separate project similar to say [google-java-format](https://github.com/diffplug/spotless/tree/main/plugin-gradle#google-java-format). I would think about the `README.md` for this project - what is the philosophy, what users will want it, which don't, etc.

If that's too much work and you just want to use if for your own projects, you can always put it in `buildSrc` and just call `addStep` :)

**Comment by Pankraz76** [comment]
Thanks for the insight. I partially agree.

If this step is generic, it should apply to Google, Palantir, and any other Java formatter. The same goes for wildcard imports—this is a general feature we want, regardless of the formatter in use.

But what if one project uses Palantir and wants to apply this recipe, even though it lives under the Google formatter? How would we handle that?

Assuming this is as generic as the wildcard feature, I don’t see why it should be treated differently.

**Comment by Pankraz76** [comment]
What about splitting the process?
We could first address static abstract redundancies in a `redundantDeclaration` step, and then extract the initialization logic into a `redundantInitialization` step.

This way, users can choose which aspect they want to focus on. 
The assumption is that the first step would avoid any obvious overhead, as already marked by the IDE.
